pkg/render/apiserver.go

```diff
@@ -675,7 +679,7 @@ func (c *apiServerComponent) secretsClusterRole() []client.Object {
 		&rbacv1.ClusterRole{
 			TypeMeta: metav1.TypeMeta{Kind: "ClusterRole", APIVersion: "rbac.authorization.k8s.io/v1"},
 			ObjectMeta: metav1.ObjectMeta{
-				Name: name,
+				Name: APIServerSecretsRBACName,
```

Comment:
I know this is pre-existing and not part of your changes, but I think the rules should include a list of valid resource names that the API server is allowed to "get" - we can know this by looking at the full set of namespaces that have tenants in it (which should be passed in from the controller as an argument). Otherwise, the apiserver suddenly has access to any secret in the cluster.

We do a similar thing for cluster role bindings for Linseed: https://github.com/tigera/operator/blob/09be800995e4745154c9cff1250b3ff623be994b/pkg/controller/logstorage/linseed/linseed_controller.go#L365-L369

Feel free to defer this to another PR / a backlog item though - I won't make you fix my own flawed code as part of this one :grin:

Reply:
I think we can get the secret name from the managementcluster resource. From what I see, linseed uses these namespaces to bind to multiple service accounts. But since there will always be a single API server for all tenants, not sure we need it.